core/spec.md

```diff
@@ -91,51 +91,55 @@ For easy reference, the JSON serialization of a Registry adheres to this form:
         "strict": BOOL, ?               # Just "enum" values or not. Def=true
-        "required": BOOL, ?             # default: false, from a CLI POV?
-        "attributes": { ... }, ?        # only if "type" == "object"
-        "itemType": "TYPE", ?           # only if "type" == "array"
-
-        "ifValue": {                    # only if "type" != "object"
-          VALUE: {                      # possible attribute value
-            "parentAttributes": { ... } # see "attributes" above (siblings)
+        "required": BOOL, ?             # Default: false, from a CLI POV?
+        "attributes": { ... }, ?        # Only if "type" == "object"
+        "keyType": "TYPE", ?            # Only if "type" == "map"
+        "itemType": "TYPE", ?           # Only if "type" == "array" or "map"
+
+        "ifValue": {                    # Only if "type" != "object"
+          VALUE: {                      # Possible attribute value
+            "siblingAttributes": { ... } # See "attributes" above
           } *
         } ?
       } *
     },
 
-    "groups": [
-      { "singular": "STRING",           # eg. "endpoint"
+    "groups": {
+      "STRING": {                       # Plural name, eg. "endpoints"
         "plural": "STRING",             # eg. "endpoints"
-        "attributes": { ... }, ?        # see "attributes" above
+        "singular": "STRING",           # eg. "endpoint"
+        "attributes": { ... }, ?        # See "attributes" above
 
-        "resources": [
-          { "singular": "STRING",       # eg. "definition"
+        "resources": {
+          "STRING": {                   # Plural name, e.g. "definitions"
             "plural": "STRING",         # eg. "definitions"
-            "versions": UINT ?          # num Versions(>=0). Def=1, 0=unlimited
-            "versionId": BOOL, ?        # Supports client specific Version IDs
+            "singular": "STRING",       # eg. "definition"
+            "versions": UINT ?          # Num Versions(>=0). Def=1, 0=unlimited
+            "versionId": BOOL, ?        # Supports client specified Version IDs
             "latest": BOOL, ?           # Supports client "latest" selection
             "hasDocument": BOOL, ?      # Has a separate document. Def=true
-            "attributes": { ... } ?     # see "attributes" above
+            "attributes": { ... } ?     # See "attributes" above
           } *
-        ] ?
+        } ?
       } *
-    ] ?
+    } ?
   } ?
 
   # Repeat for each Group type
   "GROUPsUrl": "URL",                              # eg. "endpointsUrl"
   "GROUPsCount": INT                               # eg. "endpointsCount"
-  "GROUPs": {                                      # only if inlined
-    "ID": {                                        # the Group id
-      "id": "STRING",                              # a Group
+  "GROUPs": {                                      # Only if inlined
+    "ID": {                                        # The Group id
```

Comment:
This comment and the next line look like they're the wrong way round. Or at least odd.